src/c/text/retezce.md

```diff
@@ -1,2 +1,198 @@
+zapisovat, lze jej pouze číst[^2].
+2) Samotný výraz literálu se při běhu programu vyhodnotí jako adresa prvního znaku řetězce
+uloženého v globální paměti.
+3) Datový typ literálu bude
+[ukazatel na konstantní znak](../prace_s_pameti/ukazatele.md#konstantní-ukazatele), tedy
+`const char*`. Tento datový typ říká, že hodnotu znaku na dané adrese nelze měnit.
+
+[^2]: Tyto řetězce jsou pouze pro čtení zejména z toho důvodu, aby je šlo sdílet. Pokud například
+v programu použijete třikrát stejný řetězcový literál, překladač může v paměti pole pro tento
+literál vytvořit pouze jednou, aby ušetřil paměť. Kvůli toho ale musí být řetězce pouze pro čtení,
+pokud bychom totiž takto sdílený řetězec změnili, změnilo by to i hodnotu všech ostatních literálů,
+které se vyhodnotí na jeho adresu, což by bylo dost neintuitivní.
+
+Pomocí řetězcového literálu si tak můžeme značne usnadnit zápis řetězců v programech, jelikož
+nemusíme přemýšlet nad délkou pole, nemusíme pamatovat na umístění znaku `'\0'` na konec řetězce
+a ani nemusíme obalovat jednotlivé znaky do apostrofů:
+```c,editable,mainbody
+#include <stdio.h>
+
+int main() {
+    const char* text = "UPR";
+    printf("%s\n", text);
+    return 0;
+}
+```
+Je však třeba pamatovat na to, že takto vytvořené řetězce jsou opravdu pouze pro čtení, a nesmíme
+tak do nich zapisovat. Pokud je budete ukládat do proměnné, tak použijte datový typ `const char*`,
+díky kterému vás překladač bude hlídat, abyste se do takovéhoto řetězce omylem nesnažili něco zapsat.
+
+Pokud byste chtěli použít řetězcový literál pro vytvoření řetězce, který lze měnit, můžete ho uložit
+do proměnné typu `char[]` (tj. pole znaků, které lze měnit):
+```c,editable,mainbody
+#include <stdio.h>
+
+int main() {
+    char text[] = "UPR";
+    text[0] = 'A';
+    printf("%s\n", text);
+    return 0;
+}
+```
+
+> Pokud jsou vám řetězcové literály povědomé, je to kvůli toho, že jsme je již mnohokrát
+> využili při volání funkce `printf`.
+
+### K čemu slouží nulový znak na konci?
+U polí je trochu nepraktické to, že pokud je chceme poslat do nějaké funkce, musíme spolu s
+ukazatelem na první prvek pole předat také jeho
+[velikost](../pole/staticke_pole.md#předávání-velikosti-pole), aby funkce věděla, ke kolika prvkům
+si může dovolit přistoupit. Jiným způsobem, jak určit velikost pole, je zvolit si speciální hodnotu,
+která bude značit konec pole. Když kód, který s takovýmto polem bude pracovat, na tuto speciální
+hodnotu narazí, tak bude vědět, že dále v paměti již pole nepokračuje.
+
+Tento mechanismus je využit právě u řetězců zakončených nulou, kde onou speciální hodnotou je právě
+tzv. `NUL` znak, který má číselnou hodnotu `0`. Například při procházení řetězce v cyklu tak nemusíme
+dopředu znát jeho délku, stačí cyklus ukončit, jakmile narazíme na znak `'\0'`. Například funkce
+pro spočtení délky řetězce by mohla vypadat takto[^3]:
+```c
+int delka_retezce(const char* retezec) {
+    int delka = 0;
+    while (*retezec != '\0') {
```

Comment:
neni to moc magic s ukazateli?
nebo nejak zmenit, ze si tu adresu posouvame dal a dal?